src/main/kotlin/org/rust/lang/core/psi/RsPsiFactory.kt

```diff
@@ -10,6 +10,10 @@ import org.rust.lang.core.psi.impl.mixin.valueParameters
 import org.rust.lang.core.psi.util.childOfType
 
 class RsPsiFactory(private val project: Project) {
+    fun createSelf(): RsSelfParameter =
+        createFromText<RsFunction>("fn main(self){}")?.selfParameter
+            ?: error("Failed to create self element")
```

Comment:
I think ideally we should let the user chose between `self`, `&self` and `&mut self`, but as a basic implementation I would expect `&self` to be added, it seems much more common. What do you think?